explore/hash.h

```diff
@@ -57,7 +57,7 @@ namespace MURMUR_HASH_3
 //-----------------------------------------------------------------------------
 // Finalization mix - force all bits of a hash block to avalanche
 
-static inline uint32_t fmix(uint32_t h)
+static inline uint32_t fmix(uint32_t h) noexcept
```

Comment:
Is there a reason why this is not constexpr?

Reply:
In c++11 constexpr functions are **very** restricted. The function body can only contain a return statement and no other statements. This function is 100% constexpr but only when we can use the relaxed rules introduced in c++14

Reply:
In c++11 constexpr functions are very restricted. The function body can only contain a return statement and no other statements. This function is 100% constexpr but only when we can use the relaxed rules introduced in c++14